### PR TITLE
[TST] make tools.linalg __main__ section into tests

### PR DESCRIPTION
**Pull request by jbrockmendel**
Both the module and the test file are added to the fully-lint list.

### CODE REVIEW

File: statsmodels/tools/tests/test_linalg.py [hunk 1]
```diff
@@ -21,3 +22,25 @@ def test_stationary_solve_2d():
...
+    # Note on Windows32:
+    #    linalg doesn't always produce the same results in each call
+    import scipy.linalg
+    a0 = np.random.randn(100, 10)
+    b0 = a0.sum(1)[:, None] + np.random.randn(100, 3)
+
+    # smoke-tests for lstsq, pinv, pinv2
+    linalg.lstsq(a0, b0)
+    linalg.pinv(a0)
+    linalg.pinv2(a0)
```
Comment (josef-pkt) [inline statsmodels/tools/tests/test_linalg.py]: pinv and pinv2 are called again below, so there is no need to smoke test them

lstsq could get the same assert against scipy.linalg.lstsq as the other two

**Comment by josef-pkt** [comment]
looks fine to me, after adjusting the unit tests
it leaves it open whether or how long we want to keep the scipy duplicates

File: statsmodels/tools/tests/test_linalg.py (hunk 1 above)
Comment (jbrockmendel) [reply statsmodels/tools/tests/test_linalg.py]: Sounds good, will update.

**Comment by jbrockmendel** [comment]
> it leaves it open whether or how long we want to keep the scipy duplicates

anybody want to take bets on my opinion?

**Comment by josef-pkt** [comment]
@jbrockmendel  thank you

**Pull request merged**